doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,74 @@
+# Boundary Sync REST API Design
+
+## Context
+
+Based on the [data model](./adr-001-data-models.md), the following REST API design is proposed to access it:
+
+### Authentication
+
+| Path            | Method | Body      | Authentication | Responses                                             | Notes                                        |
+| --------------- | ------ | --------- | -------------- | ----------------------------------------------------- | -------------------------------------------- |
+| `/auth/login/`  | GET    | -         | Allow Any      | 200 OK,<br />401 Not Authorized                       | Returns loging JSON if current session valid |
```

Comment:
This should return at least the Role of the user, so decisions can be made in the front-end. It may also return the user's email, which can be shown in the UI as the currently logged in user. We don't have a spot for this in the UI, but most apps do, and we may well have one at some point.